src/ctap2/mod.rs

```diff
@@ -809,5 +809,440 @@ pub(crate) fn set_or_change_pin_helper(
     };
     // the callback is expecting `Result<(), AuthenticatorError>`, but `ChangeExistingPin`
     // and `SetNewPin` return the default `ClientPinResponse` on success. Just discard it.
-    callback.call(res.map(|_| ()));
+    callback.call(res.map(|_| T::from(())));
+}
+
+pub(crate) fn bio_enrollment(
+    dev: &mut Device,
+    command: BioEnrollmentCmd,
+    status: Sender<crate::StatusUpdate>,
+    callback: StateCallback<crate::Result<crate::ManageResult>>,
+    alive: &dyn Fn() -> bool,
+) -> bool {
+    let authinfo = match dev.get_authenticator_info() {
+        Some(i) => i.clone(),
```

Comment:
I don't see why we need to clone the authenticator info here.

Reply:
Oh, indeed. `clone()` is unnecessary.